xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/Inputstream.h

```diff
@@ -258,7 +270,7 @@ namespace addon
     * @param enable true for enable, false for disable
     * @remarks
     */
-    virtual void EnableStream(int streamid, bool enable) { }
+    virtual void EnableStream(int streamid, bool enable) = 0;
```

Comment:
Because you implement logic here (not especially the one above but the other two where you memset things), should be part of the addon developer.